libraries/chain/include/xgt/chain/database.hpp

```diff
@@ -134,6 +126,57 @@ namespace xgt { namespace chain {
             TBenchmark benchmark = TBenchmark(0, []( uint32_t, const abstract_index_cntr_t& ){});
          };
 
+         struct transaction_processor
+         {
+            struct tx_comparator
+            {
+               bool operator()( const std::tuple<fc::time_point_sec, transaction_id_type>& a, const std::tuple<fc::time_point_sec, transaction_id_type>& b) const
+               {
+                  if ( std::get<0>(a) < std::get<0>(b) )
+                     return true;
+                  if ( std::get<0>(a) == std::get<0>(b) )
+                     if (std::get<1>(a) < std::get<1>(b) )
+                        return true;
+                  return false;
+               }
+            };
+
+            bool _is_producing = false;
+            bool _is_pending_tx = false;
+            map< std::tuple<fc::time_point_sec, transaction_id_type>, signed_transaction, tx_comparator > _popped_tx;
```

Comment:
If tx appears on different block, will it satisfy uniqueness? Break up data structures?